### PR TITLE
docs: update contextIsolation documentation regarding globals.

### PR DESCRIPTION
**Pull request by bcpete**
#### Description of Change
Updated documentation regarding contextIsolation and access to the global window and document variables. I also added alternatives. 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines]

#### Release Notes
Notes: Update to contextIsolation documentation

### CODE REVIEW

File: docs/api/browser-window.md [hunk 1]
```diff
@@ -337,11 +337,12 @@ It creates a new `BrowserWindow` with native properties as set by the `options`.
       more details.
     * `contextIsolation` Boolean (optional) - Whether to run Electron APIs and
       the specified `preload` script in a separate JavaScript context. Defaults
-      to `false`. The context that the `preload` script runs in will still
-      have full access to the `document` and `window` globals but it will use
+      to `false`. The context that the `preload` script runs in will not
+      have full access to the `document` and `window` globals, will use
```
Comment (nornagon) [inline docs/api/browser-window.md]: This isn't accurate. They have access to the `document` and `window` objects, but they don't see the _same_ objects as the main content.

Think of it this way: in C++, there's a `Document` (and `Window`) object. To expose it to JavaScript, we create a wrapper object that translates JavaScript calls into C++ calls. When `contextIsolation` is on, there are _two_ wrapper objects: one for the preload script and one for the main content. Setting values on one `window` wrapper object doesn't make those values appear on the other wrapper object.

That isn't quite how it actually works, but it's a good enough analogy for this case.

Comment (felixrieseberg) [reply docs/api/browser-window.md]: @bcpete I feel like you could copy-edit @nornagon's explanation and just add _that_ to the docs 😉

Comment (bcpete) [reply docs/api/browser-window.md]: I understand, I feel like the wording is misleading. When I read "full access to the document and window globals" it makes me think I can alter the globals for the main content by mutating the window object within the preload script. Like you pointed out though this is not the case as they have different contexts.

Which was the whole point of adding contextIsolation I'd assume, it's kind of in the name, I just think for anyone digging through the docs trying to find a solution to fit their use case this section is a little bit misleading.

Any suggestions on how to improve my changes? Maybe I should change the section to better describe what context isolation is doing, and point towards solutions for mutating the main contents globals?

Comment (bcpete) [reply docs/api/browser-window.md]: Sounds good, thanks for the feedback!

Comment (bcpete) [reply docs/api/browser-window.md]: Pushed some changes, let me know if you'd like anything added.

File: docs/api/browser-window.md [hunk 2]
```diff
@@ -337,11 +337,13 @@ It creates a new `BrowserWindow` with native properties as set by the `options`.
       more details.
     * `contextIsolation` Boolean (optional) - Whether to run Electron APIs and
       the specified `preload` script in a separate JavaScript context. Defaults
-      to `false`. The context that the `preload` script runs in will still
-      have full access to the `document` and `window` globals but it will use
+      to `false`. The context that the `preload` script runs will
+      have full access to the `document` and `window` globals, however these will not be
+      the same `document/window` objects in the browserWindow. The preload script will use
```
Comment (nornagon) [inline docs/api/browser-window.md]:
```suggestion
      the same `document`/`window` objects in the browserWindow—properties set
      on `window` in the preload script will not be visible to the loaded content. The
      preload script will use
```
A little repetition can't hurt :)

Comment (bcpete) [reply docs/api/browser-window.md]: 👍

**Comment by bcpete** [comment]
@felixrieseberg you got it, will do this tomorrow night 👍

**Comment by ckerr** [comment]
@bcpete thanks for the PR and also for being so open to revisions :+1:

**Comment by bcpete** [comment]
Absolutely, working on it now.

**Comment by bcpete** [comment]
@codebytere So I tried to rebase, but now there are a bunch of extra files it's saying I changed. I had merge conflicts, so I just accepted all incoming changes, but now I think that was a mistake 🤦‍♂️. Any suggestions on how to proceed? I'm sorry I am not used to working on a project being touched by this many people.

**Comment by miniak** [comment]
@bcpete do you still want to get this PR merged? Try just resetting your branch on the latest master, reverting all unrelated changes and only applying your documentation change.

**Comment by miniak** [comment]
Closing due to lack of activity

**Pull request closed without merging**